### PR TITLE
Allow changing of expired password via vic-machine debug [specific-ci=6-11-Debug]

### LINKED ISSUE #4214: Root Password expired when trying to remote access to a VCH after setting debug mode
I've enabled the debug mode, enabled ssh and setup a root password but I'm not able to login to the vic. It seems that the password is not recognized.
I've tried to send an "authorized_keys" file for my user to be able to login without password but I get asked for a password to be changed so it seems that the VCH root password expire and the debug command is not able to automatically change it.
I've also done  a reboot of the VCH.

command to setup: 
```
[] # vic-machine-linux debug --target 'vcenter'/'Datacenter' --user 'admin@domain' --password 'adminpw' --compute-resource 'Cluster'/'Resource Pool' --thumbprint 'thumbprint' --name 'vchhostname' --authorized-key authorized_keys -pw 'root' --enable-ssh
INFO[2017-03-10T12:22:42+01:00] ### Configuring VCH for debug ####           
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] VCH ID: VirtualMachine:vm-115919             
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] Installer version: v0.8.0-7315-c8ac999       
INFO[2017-03-10T12:22:43+01:00] VCH version: v0.8.0-7315-c8ac999             
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] SSH to appliance:                            
INFO[2017-03-10T12:22:43+01:00] ssh root@ip
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] VCH Admin Portal:                            
INFO[2017-03-10T12:22:43+01:00] https://ip:2378                     
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] Published ports can be reached at:           
INFO[2017-03-10T12:22:43+01:00] ip                                  
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] Docker environment variables:                
INFO[2017-03-10T12:22:43+01:00] DOCKER_HOST=ip:2375                 
INFO[2017-03-10T12:22:43+01:00]                                              
INFO[2017-03-10T12:22:43+01:00] Connect to docker:                           
INFO[2017-03-10T12:22:43+01:00] docker -H ip:2375 info              
INFO[2017-03-10T12:22:43+01:00] Completed successfully                       

[] # ssh ip
You are required to change your password immediately (password aged)
Last login: Fri Mar 10 10:48:25 2017 from -----------
Changing password for root.
(current) UNIX password: 
Connection to ip closed.
```

**VIC version:**

v0.8.0-7315-c8ac999       

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?

**Steps to reproduce:**
Not sure as the problem arise after the root password expire (not sure how it's setup)

**Actual behavior:**
When logging in with ssh using authorized_keys I get asked to change password. 
Changing password with vic-machine don't work (the password I set is not recognized).

**Expected behavior:**
Begin able to login to vic with ssh or console.

** Logs **
Running the command to setup ssh/enable authorization_keys/changing root password I get this output in the initialization & watchdog logs:
```
time="2017-03-10T11:37:14Z" level=error msg="Expected line to end with /bin/bash: root:x:0:0:root:/root:/bin/false\n" 
time="2017-03-10T11:37:14Z" level=error msg="Failed to enable shell: Expected line to end with /bin/bash: root:x:0:0:root:/root:/bin/false\n" 
time="2017-03-10T11:37:14Z" level=info msg="Attempted to start ssh service:\n %s{[] 0 [0 0 0 0] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0}" 
time="2017-03-10T11:37:14Z" level=error msg="Expected line to end with /bin/bash: root:x:0:0:root:/root:/bin/false\n" 
time="2017-03-10T11:37:14Z" level=error msg="Failed to enable shell: Expected line to end with /bin/bash: root:x:0:0:root:/root:/bin/false\n" 
```

### PR DESCRIPTION
**Pull request by hickeng**
This updates the pam config for chpasswd to allow changing of the password
when expired, but only when invoked from an already authenticated root
session.
In the case of the appliance, that means invoked programmatically by
vic-init

Change the date forwards past the password expiry:
```
root@ [ ~ ]# date +%D -s 12/12/17
root@ [ ~ ]# date
Tue Dec 12 00:02:26 UTC 2017
```
Change the password via `chpasswd` directly:
```
root@ [ ~ ]# journalctl -f
-- Logs begin at Thu 2017-04-06 22:48:35 UTC. --
Apr 06 22:48:54 Linux chsh[320]: changed user 'root' shell to '/bin/bash'
Apr 06 22:48:54 Linux systemd[1]: Started OpenSSH Daemon.
Apr 06 22:48:54 Linux chsh[325]: changed user 'root' shell to '/bin/bash'
Apr 06 22:48:54 Linux chpasswd[329]: pam_unix(chpasswd:chauthtok): password changed for root
Apr 06 22:48:54 Linux sshd[331]: Accepted keyboard-interactive/pam for root from 192.168.78.215 port 35980 ssh2
Apr 06 22:48:54 Linux sshd[331]: pam_unix(sshd:session): session opened for user root by (uid=0)
Apr 06 22:48:54 Linux sshd[331]: pam_systemd(sshd:session): Failed to connect to system bus: No such file or directory
Apr 06 22:49:04 Linux kernel: random: nonblocking pool is initialized
Dec 12 00:00:00 Linux systemd[1]: Time has been changed
Dec 12 00:00:05 Linux chpasswd[343]: pam_unix(chpasswd:chauthtok): password changed for root
```

I've not tested this directly on a VCH without interactive shell open as yet - would require setting ESX host time to more than
three months ahead of ISO dates, _or_ the build system back to author the ISO in the past.

Fixes #4214

### CODE REVIEW

**Comment by mdubya66** [comment]
Approved for 1.1

**Comment by hickeng** [comment]
Still hitting the change password requirement when using only ssh keys. The following is ssh debug trace from the client - we auth correctly but are denied until password is changed:
```
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /home/vagrant/.ssh/id_rsa
debug3: send_pubkey_test
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 60
debug1: Server accepts key: pkalg ssh-rsa blen 151
debug2: input_userauth_pk_ok: fp SHA256:0pdHtDLLlDpIXt4OcfUTWWexgvuSrhQZLNyT8LldLtA
debug3: sign_and_send_pubkey: RSA SHA256:0pdHtDLLlDpIXt4OcfUTWWexgvuSrhQZLNyT8LldLtA
debug3: send packet: type 50
debug3: receive packet: type 52
debug1: Authentication succeeded (publickey).
Authenticated to 192.168.78.127 ([192.168.78.127]:22).
debug1: channel 0: new [client-session]
debug3: ssh_session2_open: channel_new: 0
debug2: channel 0: send open
debug3: send packet: type 90
debug1: Requesting no-more-sessions@openssh.com
debug3: send packet: type 80
debug1: Entering interactive session.
debug1: pledge: network
debug3: receive packet: type 80
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug3: receive packet: type 91
debug2: callback start
debug2: fd 3 setting TCP_NODELAY
debug3: ssh_packet_set_tos: set IP_TOS 0x10
debug2: client_session2_setup: id 0
debug2: channel 0: request pty-req confirm 1
debug3: send packet: type 98
debug1: Sending environment.
debug3: Ignored env XDG_SESSION_ID
debug3: Ignored env TERM
debug3: Ignored env SHELL
debug3: Ignored env SSH_CLIENT
debug3: Ignored env DOCKER_HOST
debug3: Ignored env OLDPWD
debug3: Ignored env THUMBPRINT
debug3: Ignored env IMAGE_STORE
debug3: Ignored env SSH_TTY
debug3: Ignored env VIC_NAME
debug3: Ignored env GOVC_URL
debug3: Ignored env USER
debug3: Ignored env LS_COLORS
debug3: Ignored env DATASTORE
debug3: Ignored env MAPPED_NETWORKS
debug3: Ignored env DOCKER_TLS_VERIFY
debug3: Ignored env MAIL
debug3: Ignored env PATH
debug3: Ignored env PWD
debug3: Ignored env STD_NAME
debug1: Sending env LANG = en_US.utf8
debug2: channel 0: request env confirm 0
debug3: send packet: type 98
debug3: Ignored env IPADDR
debug3: Ignored env DOCKER_CERT_PATH
debug3: Ignored env SHLVL
debug3: Ignored env HOME
debug3: Ignored env LANGUAGE
debug3: Ignored env LOGNAME
debug3: Ignored env SSH_CONNECTION
debug3: Ignored env GOPATH
debug3: Ignored env LESSOPEN
debug3: Ignored env XDG_RUNTIME_DIR
debug3: Ignored env DEBUG
debug3: Ignored env LESSCLOSE
debug3: Ignored env _
debug2: channel 0: request shell confirm 1
debug3: send packet: type 98
debug2: callback done
debug2: channel 0: open confirm rwindow 0 rmax 32768
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: PTY allocation request accepted on channel 0
debug2: channel 0: rcvd adjust 2097152
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: shell request accepted on channel 0
You are required to change your password immediately (password aged)
Changing password for root.
(current) UNIX password:
```

**Comment by hickeng** [comment]
Added `chage -d` calls to reset password expiring as well as allow changing. I think this is a better interpretation of the intent of `vic-machine debug --ssh`  that requiring a password change before SSH can succeed when an authorized key is present.

This also configures the current password to expire after 1 day - also in line with how `vic-machine debug` was intended to be used. After expiring, rerunning `vic-machine debug` will again bump the expiration date to a day in the future:
```
vagrant@devbox:~/vicsmb/src/github.com/vmware/vic$ ssh root@192.168.78.127
Warning: your password will expire in 0 days
Last login: Thu Dec  7 18:57:22 2017 from 192.168.78.215
root@ [ ~ ]#
```

**Comment by emlin** [comment]
lgtm

File: cmd/vic-init/toolbox.go [hunk 1]
```diff
@@ -99,20 +112,15 @@ func enableShell() error {
 		return err
 	}
 
+	log.Info("Attempted to enable the shell for root")
```
Comment (vburenin) [inline cmd/vic-init/toolbox.go]: I think in needs a better log message, something like: Shell for the root user has been enabled.
attempted - doesn't mean success.

Comment (hickeng) [inline cmd/vic-init/toolbox.go]: Correct - if you read the code there's no actual assurance of success.

**Comment by hickeng** [comment]
Updated with robot test - this sets the date forwards by 6 years and checks we can still update credentials via `vic-machine debug`. 6 years was chosen as it's past the support window.

File: tests/test-cases/Group6-VIC-Machine/6-11-Debug.robot [hunk 2]
```diff
@@ -16,7 +16,7 @@
 Documentation  Test 6-11 - Verify enable of ssh in the appliance
 Resource  ../../resources/Util.robot
 Suite Setup  Install VIC Appliance To Test Server
-Suite Teardown  Cleanup VIC Appliance On Test Server
+# Suite Teardown  Cleanup VIC Appliance On Test Server
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-11-Debug.robot]: uncomment

**Comment by mdubya66** [comment]
This is approved for merge into the release and is considered a stopper for 1.1. Please cherrypick.

**Comment by rajanashok** [comment]
lgtm

File: tests/test-cases/Group6-VIC-Machine/6-11-Debug.robot [hunk 3]
```diff
@@ -35,3 +35,34 @@ Enable SSH and verify
 
     # delete the keys
     Remove Files  %{VCH-NAME}.key  %{VCH-NAME}.key.pub
+
+
+*** Test Cases ***
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-11-Debug.robot]: only one test case table needed, not sure if it actually even supports two.  just eliminate this.

**Pull request merged**